libraries/AP_OpenDroneID/AP_OpenDroneID.cpp

```diff
@@ -254,12 +256,36 @@ void AP_OpenDroneID::send_location_message()
 
     const AP_GPS::GPS_Status gps_status = gps.status();
     const bool got_bad_gps_fix = (gps_status < AP_GPS::GPS_Status::GPS_OK_FIX_3D);
+    const bool armed = hal.util->get_soft_armed();
 
     Location current_location;
     if (!ahrs.get_position(current_location)) {
         return;
     }
-    const uint8_t uav_status = hal.util->get_soft_armed()? MAV_ODID_STATUS_AIRBORNE : MAV_ODID_STATUS_GROUND;
+    uint8_t uav_status = hal.util->get_soft_armed()? MAV_ODID_STATUS_AIRBORNE : MAV_ODID_STATUS_GROUND;
+    // set emergency status if chute is released
+    const auto *parachute = AP::parachute();
+    if (parachute != nullptr && parachute->released()) {
+        uav_status = MAV_ODID_STATUS_EMERGENCY;
+    }
+
+    // if we are armed with no GPS fix and we haven't specifically
+    // allowed for non-GPS operation then declare an emergency
+    if (got_bad_gps_fix && armed && !option_enabled(Options::AllowNonGPSPosition)) {
+        uav_status = MAV_ODID_STATUS_EMERGENCY;
+    }
+
+    // if we are disarmed and falling at over 3m/s then declare an
+    // emergency. This covers cases such as deliberate crash with
+    // advanced failsafe and an unintended reboot or in-flight disarm
+    if (!got_bad_gps_fix && !armed && gps.velocity().z > 3.0) {
```

Comment:
`WPNAV_SPEED_DN` looks like it can be configured from 100 to 500 cm/s based on ArduPilot allowed range. We configure it to 225 cm/s (2.5 cm/s) so this velocity _should_  theoretically only reach 2.25m/s. However, if we configure it to 500 m/s, then wouldn't a standard landing meet the criteria and trigger an emergency declaration?

Reply:
I wonder if instead of hard-coding 3 m/s we should use `gps.velocity().z > max(wpnav_speed_dn, land_speed_high)`

Reply:
One of the other criteria is that the aircraft is not armed. Therefore, this should not trigger during a standard landing.

Reply:
In regards to using `max(wpnav_speed_dn, land_speed_high)` instead of hard coding 3, I think that this calculation needs to consider the fact that the aircraft is falling rather than landing. I don't think that the descent rate with props off and parachute deployed should be compared to a controlled landing speed. However, it would be worth looking into determining what a good threshold for our aircraft is and having the ability to set an AP param with this speed to compare to

Reply:
I agree. I used those parameters as an example because I missed the fact that this could not be a controlled landing descent. Those parameters don't make sense for the situation anymore. Do we know where 3 came from? @tridge is there a parameter we can use in place of the hard-coded value that would make sense?

Reply:
note that this threshold is only used when disarmed, so won't trigger on normal flight. It is just there to prevent triggering when on the ground

Reply:
@bhiln we certainly could set a param, or perhaps just this:
```
#ifndef ODID_FREEFALL_THRESHOLD
#define ODID_FREEFALL_THRESHOLD 3.0
# endif
````
and set it in hwdef.dat?